lib/RestructureCFG/SelectScopePass.cpp

```diff
@@ -0,0 +1,261 @@
+//
+// Copyright rev.ng Labs Srl. See LICENSE.md for details.
+//
+
+#include "llvm/ADT/DepthFirstIterator.h"
+#include "llvm/ADT/PostOrderIterator.h"
+#include "llvm/ADT/STLExtras.h"
+#include "llvm/ADT/SetVector.h"
+#include "llvm/ADT/SmallPtrSet.h"
+#include "llvm/IR/BasicBlock.h"
+#include "llvm/IR/IRBuilder.h"
+#include "llvm/IR/LLVMContext.h"
+#include "llvm/IR/Module.h"
+#include "llvm/Support/GenericDomTree.h"
+
+#include "revng/ADT/ReversePostOrderTraversal.h"
+#include "revng/RestructureCFG/ScopeGraphGraphTraits.h"
+#include "revng/RestructureCFG/ScopeGraphUtils.h"
+#include "revng/RestructureCFG/SelectScopePass.h"
+#include "revng/Support/Debug.h"
+#include "revng/Support/GraphAlgorithms.h"
+#include "revng/Support/IRHelpers.h"
+
+using namespace llvm;
+
+// Debug logger
+Logger<> SelectScopePassLogger("select-scope");
```

Comment:
If you give a shorter name to this logger, such as `Log`, all the logging code will become much more readable, because it will fit on a single line much more often.

Reply:
I changed the name of the logger here https://github.com/revng/revng/blob/a3b1cbac550057e016f37c8f9eb201e8ab290b82/lib/RestructureCFG/SelectScopePass.cpp#L27

Reply:
Wait, I lied, this causes big troubles, because there is already a `Logger` called `Log` in here https://github.com/revng/revng/blob/f780df8d6acf453cace81f95b94fd19d2ef1831f/lib/ValueMaterializer/DataFlowGraph.cpp#L19C17-L19C40, and at runtime when loading both the libraries things go bad.

I'll give it a simpler name but different from `Log`

Reply:
Ok, it is now name `ScopeLog` so it doesn't cause problems with `clopt` global bindings anymore https://github.com/revng/revng/blob/f6e301a2cf0c06a6afdcafbb365ed3fb38279627/lib/RestructureCFG/SelectScopePass.cpp#L27

Reply:
Weird. I'm sure `Log` is used as a name for a logger in many places.

In any case, the problem should go away if you make this `Logger` `static`. Please do.

Reply:
Probably they are all `static`, I didn't investigate fully but the stacktrace I got mentioned a clash caused in `clopt` during a `dlopen`, so I assumed it was some clashing related problem.

Anyway I made it `static` here and the problem disappeared https://github.com/revng/revng/blob/9208e1d0d7ecf56a86e3f44debb589c8eded149c/lib/RestructureCFG/SelectScopePass.cpp#L27